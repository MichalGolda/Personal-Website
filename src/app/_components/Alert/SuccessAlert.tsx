import Alert from "./Alert";
import type { AlertProps } from "./Alert";

export default function SuccessAlert({ message }: AlertProps) {
  return <Alert className="bg-success" message={message} />;
}
