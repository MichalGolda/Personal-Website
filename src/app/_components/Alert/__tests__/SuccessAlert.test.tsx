import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import SuccessAlert from "../SuccessAlert"

describe("SuccessAlert component", () => {
    it("renders without errors", () => {
        render(<SuccessAlert message="test message" />)

        const alert = screen.getByTestId("alert")
        const alertMessage = screen.getByText("test message")

        expect(alert).toBeInTheDocument()
        expect(alert).toHaveClass("bg-success")
        expect(alertMessage).toBeInTheDocument()
    })

    it("renders unchanged", () => {
        const { container } = render(<SuccessAlert message="test message" />)
        
        expect(container).toMatchSnapshot()
    })
})
