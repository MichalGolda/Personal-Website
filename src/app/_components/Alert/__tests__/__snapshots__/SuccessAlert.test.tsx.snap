// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessAlert component renders unchanged 1`] = `
<div>
  <div
    class="w-full p-4 text-white bottom-0 left-0 font-bold text-sm md:text-base text-center fixed z-[999]  bg-success"
    data-testid="alert"
  >
    <p>
      test message
    </p>
  </div>
</div>
`;
